open_spiel/games/dou_dizhu/dou_dizhu_utils.cc

```diff
@@ -0,0 +1,860 @@
+// Copyright 2022 DeepMind Technologies Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "open_spiel/abseil-cpp/absl/strings/str_format.h"
+#include "open_spiel/games/dou_dizhu/dou_dizhu_utils.h"
+
+
+
+namespace open_spiel{
+namespace dou_dizhu{
+
+
+// dropping suit information
+int CardToRank(int card){
+  if(card == kNumCards - 2 || card == kNumCards - 1) {
+    return card-kNumCards+kNumRanks;}
+  return card % (kNumRanks - 2);
+}
+
+
+std::string RankString(int rank){
+  if(rank < kNumRanks - 2) return std::string(1, kRankChar[rank]);
+  else if(rank == kNumRanks - 2) return "(BWJ)";
+  else if(rank == kNumRanks - 1) return "(CJ)";
+  else SpielFatalError("Non valid rank");
+}
+
+std::string FormatSingleHand(const std::array<int, kNumRanks>& hand){
```

Comment:
So, this is an example of something that should be a Span<const int>. It should convert automatically.